onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -250,6 +250,11 @@ TensorrtLogger& GetTensorrtLogger() {
   return trt_logger;
 }
 
+std::unique_lock<OrtMutex> TensorrtExecutionProvider::GetApiLock() const {
```

Comment:
we need to be careful if we expand usage of this beyond the current few locking calls since it's singleton and shared across all sessions.

Reply:
If we keep the lock on all the time, we should remove the TRT option force_sequential_engine_build, which provided an option to turn the lock off before.

Reply:
be careful when removing options from structs and api's though.

Reply:
legacy [OrtTensorRTProviderOptions](https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/session/onnxruntime_c_api.h#L466) struct: I don't think we can remove this field from the struct and build ORT directly. User application built with the old struct may fail to access force_sequential_engine_build field. We can disable this field instead?
new [OrtTensorRTProviderOptionsV2](https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/providers/tensorrt/tensorrt_provider_options.h#L13): I think we can remove this field since it's a opaque struct.

Reply:
the option has become a no-op , so it's not as critical I think
we should remove it from our documentation though